servers/cu/src/domain/client/ao-process.js

```diff
@@ -286,6 +292,11 @@ const processDocSchema = z.object({
 
 function latestCheckpointBefore (destination) {
   return (curLatest, checkpoint) => {
+    /**
+     * What we will compare. In some cases, the timestamp is in the evaluation
+     * key of the checkpoint object. This handles those cases.
+     */
+    const evaluation = checkpoint.evaluation ?? checkpoint
```

Comment:
What about the other fields that are compared? `ordinate` and `cron`?

Reply:
Any way to for `latestCheckpointBefore`'s `reduce` to have a single api, instead of supporting two different input shapes? Any way to adapt the caller to achieve that? Think the [open-closed principle](https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle)